contracts/core/fee-calculators/PerformanceFeeCalculator.sol

```diff
@@ -0,0 +1,312 @@
+        ScaleValidations.validateMultipleOfBasisPoint(parameters.streamingFeePercentage);
+
+        require(
+            parameters.highWatermarkResetFrequency >= parameters.profitFeeFrequency,
+            "PerformanceFeeCalculator.validateFeeParameters: Fee collection frequency must exceed highWatermark reset."
+        );
+    }
+
+    function calculateStreamingFee()
+        internal
+        view
+        returns(uint256)
+    {
+        uint256 timeSinceLastFee = block.timestamp.sub(lastStreamingFeeTimestamp());
+
+        return timeSinceLastFee.mul(streamingFeePercentage()).div(ONE_YEAR_IN_SECONDS);
+    }
+
+    function calculateProfitFee(uint256 _streamingFee)
+        internal
+        view
+        returns(uint256)
+    {
+        uint256 rebalancingSetValue = calculateRebalancingSetValue(msg.sender);
+        uint256 postStreamingValue = rebalancingSetValue.mul(ONE_HUNDRED_PERCENT.sub(_streamingFee)).deScale(); 
+        uint256 highWatermark = highWatermark();
+
+        uint256 gainedValue = postStreamingValue > highWatermark ? postStreamingValue.sub(highWatermark) : 0;
+
+        return gainedValue.mul(profitFeePercentage()).div(rebalancingSetValue);
+    }
+
+    function calculateRebalancingSetValue(
+        address _rebalancingSetTokenAddress
+    )
+        internal
+        view
+        returns (uint256)
+    {
+        IRebalancingSetTokenV2 rebalancingSetToken = IRebalancingSetTokenV2(_rebalancingSetTokenAddress);
+
+        uint256 unitShares = rebalancingSetToken.unitShares();
+        uint256 naturalUnit = rebalancingSetToken.naturalUnit();
+        ISetToken currentSet = rebalancingSetToken.currentSet();
+
+        uint256 collateralValue = SetUSDValuation.calculateSetTokenDollarValue(
+            currentSet,
+            oracleWhiteList
+        );
+
+        return collateralValue.mul(unitShares).div(naturalUnit);
+    }
+
+    function updateFeeState(
+        uint256 _streamingFee,
+        uint256 _profitFee
+    )
+        internal
+    {
+        feeState[msg.sender].lastStreamingFeeTimestamp = block.timestamp;
+
+        uint256 rebalancingSetValue = calculateRebalancingSetValue(msg.sender);
+        uint256 postStreamingValue = rebalancingSetValue.mul(ONE_HUNDRED_PERCENT.sub(_streamingFee)).deScale(); 
+        if (_profitFee > 0) {
+            feeState[msg.sender].lastProfitFeeTimestamp = block.timestamp;
+            feeState[msg.sender].highWatermark = postStreamingValue;
+        }
+
+        if (block.timestamp.sub(lastProfitFeeTimestamp()) >= highWatermarkResetFrequency()) {
+            feeState[msg.sender].highWatermark = postStreamingValue;
+        }
+    }
+
+    function exceedsProfitFeeFrequency() internal view returns (bool) {
+        return block.timestamp.sub(lastProfitFeeTimestamp()) > profitFeeFrequency();
+    }
+
+    function lastStreamingFeeTimestamp() internal view returns (uint256) {
+        return feeState[msg.sender].lastStreamingFeeTimestamp;
```

Comment:
For more flexibility and potential reusability, I prefer passing in the setAddress for each of these getters. 

The main usecase is that we may want to expose external getter functions to retrieve these figures.